Mobile Buy SDK/Mobile Buy SDK/Additions/NSDateFormatter+BUYAdditions.m

```diff
@@ -38,8 +38,29 @@ + (NSDateFormatter*)dateFormatterForShippingRates
 + (NSDateFormatter*)dateFormatterForPublications
 {
 	NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+	
+	if (![self is24hTimeFormat]) {
+		[dateFormatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_GB"]];
+	}
+	
 	dateFormatter.dateFormat = @"yyyy-MM-dd'T'HH:mm:ssZ";
 	return dateFormatter;
 }
 
++ (BOOL)is24hTimeFormat
+{
+	NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+	[formatter setLocale:[NSLocale currentLocale]];
+	[formatter setDateStyle:NSDateFormatterNoStyle];
+	[formatter setTimeStyle:NSDateFormatterShortStyle];
+	
+	NSString *dateString = [formatter stringFromDate:[NSDate date]];
+	
+	NSRange amRange = [dateString rangeOfString:[formatter AMSymbol]];
+	NSRange pmRange = [dateString rangeOfString:[formatter PMSymbol]];
+	BOOL is24h = (amRange.location == NSNotFound && pmRange.location == NSNotFound);
+	
+	return is24h;
```

Comment:
It is not checking locale, it is checking user settings for 24 hour format - I don't know if it is possible to check without it. But I think you are right that it is possible not to do it at all, you can just set some locale (for example "en_GB") insteat or users current and the date object will be ok. To make sure that you understand the problem: set region as United Kingdom and 12 hour format. In this case all dates (coming from your backend) that have HH more than 11 would not be initialized